compaction/task_manager_module.cc

```diff
@@ -423,6 +423,7 @@ future<> table_major_keyspace_compaction_task_impl::run() {
 
 future<> cleanup_keyspace_compaction_task_impl::run() {
     co_await _db.invoke_on_all([&] (replica::database& db) -> future<> {
+        co_await db.flush_all_tables(replica::database::wait_for_commitlog_reclaim::yes);
```

Comment:
Currently, `nodetool compact` invokes the `keyspace_cleanup` api for all non-local keyspaces.
This will cause `flush_all_tables` to be called repeatedly for each keyspace.
If the system is idle/read-only this will just consume extra cpu cycles to check all tables and all compaction_groups, but if there is some write workload, this will cause a flurry of small sstables as memtables would be flushed in quick succession.
As optimizations (can be done as a follow up), we can:
1. adapt `maybe_flush_all_tables` to prevent that so it will flush all tables only once and then wait for some period.
2. Introduce a global flush api and call it from scylla-nodetool and do first flush_all_tables there.
3. Instead of flushing all tables, get a list of tables (or better compaction groups if possible) that need flush, based on the currently held commitlog segments (after forcing a new active commitlog segment)

Reply:
@bhalevy hi Benny,  the java-based `nodetool compact` does invokes the `keyspace_cleanup` for all non-local keyspaces if the keyspace is not specified in command line. but our `scylla-nodetool compact` calls into `global_major_compaction_task_impl::run()`, which does uses `maybe_flush_all_tables()` by default, and it is called for a single time.

do you think we should optimize for the java-based tool?

Reply:
No, I meant implementing a similar optimization for cleanup.